### PR TITLE
Automatic word-break in highlighted text, combine_adjacent support

### PR DESCRIPTION
**Pull request by aliabid94**
Support word break in highlighted text, and allow combining adjacent similar categories.

Screen Shot 2022-05-10 at 4 22 05 PM

### CODE REVIEW

**Comment by pngwn** [comment]
The spacing seems a little off here.

File: ui/packages/highlighted-text/src/HighlightedText.svelte [hunk 1]
```diff
@@ -104,17 +104,18 @@
 		</div>
 	{/if}
 	<div
-		class="textfield bg-white dark:bg-transparent rounded-sm box-border max-w-full break-word inline-flex flex-wrap gap-1"
+		class="textfield bg-white dark:bg-transparent rounded-sm box-border max-w-full break-word leading-8"
```
Comment (pngwn) [inline ui/packages/highlighted-text/src/HighlightedText.svelte:107]:
```suggestion
		class="textfield bg-white dark:bg-transparent rounded-sm box-border max-w-full break-word leading-7"
```

**Review by pngwn — COMMENTED** [review]
Didn't want to push to your branch while it might be in progress. PRed some css tweak against this branch: https://github.com/gradio-app/gradio/pull/1213

**Comment by pngwn** [comment]
With recent CSS tweaks: 
Screenshot 2022-05-11 at 19 43 06
Screenshot 2022-05-11 at 19 42 31

**Comment by abidlabs** [comment]
Looks great, works great!

**Review by abidlabs — APPROVED** [review]

**Pull request merged**